### PR TITLE
Fire `onResolveRemoteAuthority` if authority exists in init data

### PR DESCRIPTION
**Pull request by joyceerhl**

### CODE REVIEW

File: src/vs/workbench/api/common/extHostExtensionService.ts [hunk 1]
```diff
@@ -699,6 +700,21 @@ export abstract class AbstractExtHostExtensionService extends Disposable impleme
...
+					return;
+				}
+
+				return (
+					this._activateById(desc.identifier, { startup: true, extensionId: desc.identifier, activationEvent: `onResolveRemoteAuthority:${this._initData.remote.authority}` })
+						.then(undefined, err => this._logService.error(err))
+				);
+			})
+		).then(() => { });
+	}
```
Comment (alexdima) [inline src/vs/workbench/api/common/extHostExtensionService.ts:716]: Could we replace this whole block with `return this._activateByEvent(`onResolveRemoteAuthority:${this._initData.remote.authority}`, false);` ?

File: src/vs/workbench/api/common/extHostExtensionService.ts [hunk 2]
```diff
@@ -653,7 +653,8 @@ export abstract class AbstractExtHostExtensionService extends Disposable impleme
 		this._register(this._extHostWorkspace.onDidChangeWorkspace((e) => this._handleWorkspaceContainsEagerExtensions(e.added)));
 		const folders = this._extHostWorkspace.workspace ? this._extHostWorkspace.workspace.folders : [];
 		const workspaceContainsActivation = this._handleWorkspaceContainsEagerExtensions(folders);
-		const eagerExtensionsActivation = Promise.all([starActivation, workspaceContainsActivation]).then(() => { });
+		const remoteResolverActivation = this._handleRemoteResolverEagerExtensions();
```
Comment (alexdima) [inline src/vs/workbench/api/common/extHostExtensionService.ts:656]: Could we move this to be the first line in the file? Maybe that would trigger the activation process for these extensions a bit earlier than the `*` ones.

**Review by alexdima — COMMENTED** [review]
Nice! I left a couple of comments

File: src/vs/workbench/api/common/extHostExtensionService.ts (hunk 2 above)
Comment (joyceerhl) [reply src/vs/workbench/api/common/extHostExtensionService.ts:656]: Is the intention for this to serve as the activation mechanism when the resolver is involved in setting up the remote connection as well? If it's just in the case where the resolver is not involved, I'm not sure it deserves higher priority.

**Review by alexdima — DISMISSED** [review]
Nice! But the CI failed with a left-over import.

**Review by karrtikr — APPROVED** [review]

**Pull request merged**